lib/migration.js

```diff
@@ -295,12 +296,33 @@ function mixinMigration(PostgreSQL) {
     const self = this;
     const modelDef = this.getModelDefinition(model);
     const prop = modelDef.properties[propName];
+    let result = self.columnDataType(model, propName);
+
+    // checks if dataType is set to uuid
+    let postgDefaultFn;
+    let postgType;
+    const postgSettings = prop.postgresql;
+    if (postgSettings && postgSettings.dataType) {
+      postgType = postgSettings.dataType.toUpperCase();
+    }
+
     if (prop.generated) {
-      return 'SERIAL';
+      if (result === 'INTEGER') {
+        return 'SERIAL';
+      } else if (postgType === 'UUID') {
+        if (postgSettings && postgSettings.defaultFn && postgSettings.extension) {
+          // if user provides their own extension and function
+          postgDefaultFn = postgSettings.defaultFn;
+          return result + ' NOT NULL' + ' DEFAULT ' + postgDefaultFn;
+        }
+        return result + ' NOT NULL' + ' DEFAULT uuid_generate_v4()';
+      } else {
+        console.log(chalk.red('>>> WARNING: ') +
+            `auto-generattion is not supported for type "${chalk.yellow(prop.type)}". \
+            Please add your own function to the table "${chalk.yellow(model)}".`);
```

Comment:
What do you mean by `add your own function`?

Reply:
functions of generating default values?